src/App.tsx

```diff
@@ -1,10 +1,57 @@
+/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, TodoError, TodosErrors, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoItem } from './components/TodoItem';
+import { FilterOptions, StatusFilter } from './components/StatusFilter';
+
+interface GetFilteredTodosFilter {
+  status: FilterOptions;
+}
+
+const getFilteredTodos = (todos: Todo[], filter: GetFilteredTodosFilter) => {
+  let filteredTodos = [...todos];
+
+  if (filter.status !== FilterOptions.All) {
+    filteredTodos = filteredTodos.filter(todo => {
+      if (filter.status === FilterOptions.Completed) {
+        return todo.completed;
+      }
+
+      return !todo.completed;
+    });
+  }
+
+  return filteredTodos;
```

Comment:
use switch case
try to get rid of let
you can pass switch case in filter callback